transaction/core/src/blockchain/block_contents.rs

```diff
@@ -33,7 +37,7 @@ impl BlockContents {
 }
 
 #[repr(transparent)]
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
 /// Hash of contents (i.e. transactions) in a block.
 pub struct BlockContentsHash<D: Digest = Blake2b256>(pub GenericArray<u8, D::OutputSize>);
```

Comment:
It might be easier if this held a [u8; 32] instead of GenericArray. We also have some macro machinery for things that look like [u8; 32], e.g.:

```
impl AsRef<[u8; 32]> for CompressedCommitment {
    fn as_ref(&self) -> &[u8; 32] {
        self.point.as_bytes()
    }
}

// Implements Ord, PartialOrd, PartialEq, Hash. Requires AsRef<[u8;32]>.
deduce_core_traits_from_public_bytes! { CompressedCommitment }

impl ReprBytes32 for CompressedCommitment {
    type Error = Error;
    fn to_bytes(&self) -> [u8; 32] {
        self.point.to_bytes()
    }
    fn from_bytes(src: &[u8; 32]) -> Result<Self, Error> {
        Ok(Self {
            point: CompressedRistretto::from_slice(src),
        })
    }
}

// Implements prost::Message. Requires Debug and ReprBytes32.
prost_message_helper32! { CompressedCommitment }
```

Reply:
If it did, then it wouldn't be possible to have it be generic over Digest, which is a thing @jcape requested when this was written.

Reply:
OK

Reply:
@mfaulk i think we should redevelop the `ReprBytes32` thing -- i threw that together really quickly, because we had so many types that were 32 bytes and i was tired of copypasting all the prost and serde code around, which will cause big problem on upgrade.

I think we should try to create a utility crate that holds `ReprBytes32` and possibly takes the `GetBytes` thing if this PR is merged: https://github.com/mobilecoinofficial/mobilecoin/pull/74
`GetBytes` and `ReprBytes32` are clearly in the same neighborhood, maybe we can make it so we only need one of them